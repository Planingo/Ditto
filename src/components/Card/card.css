.ant-card {
  width: 20em;
}

.ant-card-body {
  padding: 0 !important;
}

li > span > span > svg {
  font-size: 2em !important;
}

#download {
  color: var(--PRIMARY);
}

#cloud {
  color: var(--SECONDARY);
}

#delete {
  color: var(--RED);
}

.ant-card-head {
  margin: 0;
  padding: 0;
  display: flex;
  flex-direction: column;
  width: 100%;
}

.ant-card-head-title{   
  padding-right: 1em;
  padding-left: 1em;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  font-size: var(--title) !important;
}

.container {
  min-height: 2em;
}

.tag-container {
  display: flex;
  align-self: baseline;
  margin-left: 1em;
  margin-right: 1em;
  overflow: hidden;
}

span[role='tag'] {
  margin-bottom: 1em;
  min-width: 70px;
}

.ant-card-head-title {
  padding-bottom: 1em;
}