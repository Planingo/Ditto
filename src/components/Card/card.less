.ant-card-body {
  padding: 0 !important;
}

#download,
#cloud,
#delete {
  font-size: 2em !important;
}

#download {
  color: var(--PRIMARY);
}

#cloud {
  color: var(--SECONDARY);
}

#delete {
  color: var(--RED);
}

.ant-card-head {
  margin: 0;
  padding: 0;
}

.title-card{
  display: flex;
  flex-direction: column;
  width: 100%;
  // min-height: 100px;

  .title-container{
    display: flex;
    justify-content: center;
    align-items: center;

    h1 {
      overflow: hidden;
      max-width: 280px;
      padding-left: 20px;
      padding-right: 20px;
      white-space: nowrap;
      text-overflow: ellipsis;
      align-self: center;
      color: var(--GREY-160);
      font-weight: 600;
      font-size: 16px !important;
    }
  }
}

.container {
  min-height: 32px;
}

.tag-container {
  display: flex;
  align-self: baseline;
  padding-left: 20px;
  padding-right: 20px;
  overflow: hidden;
  max-width: 260px;

  span[role='tag'] {
    margin-bottom: 10px;
    min-width: 70px;
  }
}

.ant-card-head-title {
  padding-bottom: 6px;
}