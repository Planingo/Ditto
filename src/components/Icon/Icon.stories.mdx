import { Meta, ArgsTable } from '@storybook/addon-docs/blocks';
import * as icons from './Icon';

<Meta
  title="Atomes/Icon"
  argTypes={{
    component: {
      description:'component',
      control: {
        type: 'text',
      },
    },
    rotate: {
      description:'rotate', 
      control: {
        type: 'number'
      }
    },
    spin: {
      description:'spin', 
      control: {
        type: 'boolean'
      },
      table: {
        defaultValue: {
          summary: 'false',
        }
      }
    },
    style: {
      description:'style', 
      control: {
        type: 'object',
      }
    }
  }}
  args={{
    style:{fontSize: '4em'},
  }}
/>

### Icon

Use this playground to test the button component

<Canvas>
  <Story name="Default">
    {args => {
      return <div style={{display: "grid", gap: "20px", gridTemplateColumns: "repeat(auto-fill, minmax(180px, 1fr))"}}>
        {
          Object.entries(icons)
          .filter(([iconName]) => iconName.toUpperCase()[0] === iconName[0] && iconName !== 'IconProvider')
          .map(([iconName, Icon]) => 
              <div style={{display: "flex", flexDirection: "column", alignItems: "center"}} key={iconName} >
                <Icon {...args} />
                <p>{iconName}</p>
              </div>
          )
        }
        </div>
    }}
  </Story>
</Canvas>


<ArgsTable story="Default" />