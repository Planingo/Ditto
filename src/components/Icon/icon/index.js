export { AppstoreOutlinedDS } from './AppstoreOutlined'
export { ArrowLeftOutlined } from './ArrowLeftOutlined'
export { CalendarOutlined } from './CalendarOutlined'
export { ExperimentOutlined } from './ExperimentOutlined'
export { MailOutlined } from './MailOutlined'
export { PoweroffOutlined } from './PoweroffOutlined'
export { ScheduleOutlined } from './ScheduleOutlined'
export { SettingOutlined } from './SettingOutlined'
export { ShopOutlined } from './ShopOutlined'
export { SmileOutlined } from './SmileOutlined'
export { TagOutlined } from './TagOutlined'
export { TagsOutlined } from './TagsOutlined'
export { TeamOutlined } from './TeamOutlined'
export { UnorderedListOutlined } from './UnorderedListOutlined'
export { UserOutlined } from './UserOutlined'
export { WifiOutlined } from './WifiOutlined'