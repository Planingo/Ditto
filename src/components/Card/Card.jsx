import React from 'react';
import { Card as AntdCard, Tabs } from 'antd';
import './card.css'
import PropTypes from 'prop-types'

/**
 * 
 * Documentations andt du composant : https://ant.design/components/card/
 * 
 */
export default function Card({children, ...props }){
  return (
    <AntdCard
      {...props}
    />
  );
};

Card.propTypes = {
  /**
   * La liste d'actions s'affiche en bas de la carte
   */
  actions: PropTypes.array,
  /**
   * Clé actuelle du TabPane
   */
  activeTabKey: PropTypes.string,
  /**
   * Style à appliquer au contenu de la carte
   */
  bodyStyle: PropTypes.string,
  /**
   * Active/désactive le rendu de la bordure autour de la carte
   */
  bordered: PropTypes.bool,
  /**
   * Couverture de la carte
  */
  cover: PropTypes.node,
  /**
   * Clé initiale du TabPane actif, si activeTabKey n'est pas défini
  */
  defaultActiveTabKey: PropTypes.string,
  /**
   * Contenu à afficher dans le coin supérieur droit de la carte
  */
  extra: PropTypes.node,
  /**
   * Style à appliquer à la tête de carte
  */
  headStyle: PropTypes.string,
  /**
   * La carte se souleve lorsque vous survolez la carte
  */
  hoverable: PropTypes.bool,
  /**
  * Affiche un indicateur de chargement pendant que le contenu de la carte est récupéré
  */
  loading: PropTypes.bool,
  /**
   * Taille de la carte
  */
  size: PropTypes.oneOf(['default', 'small']),
  /**
  * Contenu supplémentaire dans la barre d'onglets
  */
  tabBarExtraContent: PropTypes.node,
  /**
   * Liste des têtes de TabPane
  */
  tabList: PropTypes.array,
  /**
   * CF tabs
  */
  tabProps: Tabs,
  /**
   * Titre de la carte
  */
  title: PropTypes.node,
  /**
   * Type de style de carte, peut être défini sur `inner` ou non défini
  */
  type: PropTypes.string,
  /**
   * Callback lors du changement d'onglet
  */
  onTabChange: PropTypes.func,
};

Card.defaultProps = {
  bordered: true,
  hoverable: true,
  loading: false,
  size: 'default',
}
