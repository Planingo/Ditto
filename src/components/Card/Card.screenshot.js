import { ReactScreenshotTest } from 'react-screenshot-test'
import React from 'react';
import Card from './Card';
import 'antd/dist/antd.css';
import { CloudUploadOutlined, DeleteOutlined, DownloadOutlined } from '@ant-design/icons';
import { Tooltip } from '../Tooltip/Tooltip';
import Tag from '../Tag/Tag';

describe("Screenshot", () => {
  const card = (
    <Card
  actions={[
    <Tooltip placement="bottom" title="Télécharger"><DownloadOutlined id="download" /></Tooltip>,
    <Tooltip placement="bottom" title="Envoyer"><CloudUploadOutlined id="cloud" /></Tooltip>,
    <Tooltip placement="bottom" title="Supprimer"><DeleteOutlined id="delete" /></Tooltip>
  ]}
  cover={
    <>
      <div className="title-card">
        <div className="container">
          <Tooltip placement="bottom" title="Métamorphose, Ancient Runes, Arithmancy">
            <div className="tag-container">
              <Tag type="lesson">Métamorphose</Tag>
              <Tag type="lesson">Ancient Runes</Tag>
              <Tag type="lesson">Arithmancy</Tag>
            </div>
          </Tooltip>
        </div>
      </div>
      <img alt="Etudiant" src="https://avatars.bugsyaya.dev/285/"/>
    </>
  }
  title="Minerva McGonagall"
/>
  );

    ReactScreenshotTest
    .create("Card")
    .viewport("Desktop", {
      width: 500,
      height: 100
    })
    .shoot("Card", card)
    .run();
})
