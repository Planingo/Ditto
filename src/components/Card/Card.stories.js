import React from 'react';

import { Card } from './Card';
import { TagsOutlined } from '../Icon/Icon'

export default {
  title: 'Components/Card',
  component: Card,
  args: {
    title: "Minerva McGonagall",
    tags: ['MÃ©tamorphose', 'Ancient Runes', 'Arithmancy'],
    icon: <TagsOutlined />,
    type: 'lesson'
  }
};

const CardGlobal = (args) => <Card {...args} />;

export const Defaut = CardGlobal.bind({});