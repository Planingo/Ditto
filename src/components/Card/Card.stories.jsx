import React from 'react';
import { Card } from './Card';
import { CloudUploadOutlined, DeleteOutlined, DownloadOutlined } from '../Icon/icon'
import { Tooltip } from '../Tooltip/Tooltip';
import { Tag } from '../Tag/Tag';
import { BrowserRouter } from 'react-router-dom'

export default {
  component: Card,
  title: 'Molécules/Card',
};

const Template = (args) => <BrowserRouter><Card {...args} /></BrowserRouter>;

const tags = [
  {id:'metamorphose', name:'Métamorphose', type: 'lesson'}, 
  {id:'runes', name:'Ancient Runes', type: 'lesson'}, 
  {id:'arithmancy', name:'Arithmancy', type: 'lesson'}]

export const Default = Template.bind({});

Default.args = {
  cover: <>
    <img src="https://avatars.bugsyaya.dev/285/" alt="Etudiant"/>
  </>,
  title: "Minerva McGonagall"
}

Default.argTypes = {
  actions: {
    control: {
      type: 'object'
    },
  },
  activeTabKey: {
    control: {
      type: 'text'
    },
  },
  bodyStyle: {
    control: {
      type: 'text'
    },
  },
  bordered: {
    control: {
      type: 'boolean'
    },
  },
  cover: {
    control: {
      type: 'object'
    },
  },
  defaultActiveTabKey: {
    control: {
      type: 'text'
    },
  },
  extra: {
    control: {
      type: 'array'
    },
  },
  headStyle: {
    control: {
      type: 'text'
    },
  },
  hoverable: {
    control: {
      type: 'boolean'
    },
  },
  loading: {
    control: {
      type: 'boolean'
    },
  },
  size: {
    control: {
      type: 'text',
      options: [
          'default',
          'small'
      ]
    },
    table: {
        defaultValue: {
            summary: 'default'
        }
    }
  },
  tabBarExtraContent: {
    control: {
      type: 'object'
    },
  },
  tabList: {
    control: {
      type: 'object'
    },
  },
  tabProps: {
    control: {
      type: 'object'
    },
  },
  title: {
    control: {
      type: 'object'
    },
  },
  type: {
    control: {
      type: 'text'
    },
  },
  onTabChange: {
    control: {
      type: 'func'
    },
  },
};
