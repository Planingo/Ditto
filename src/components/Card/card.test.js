import { TagsOutlined } from '../Icon/icon';
import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';
import React from 'react';
import { Card } from './Card';

describe('Card component', () => {
  const textContent = {
    title: "Minerva McGonagall",
    tags: ['Métamorphose', 'Ancient Runes', 'Arithmancy'],
    src: 'https://avatars.bugsyaya.dev/285/',
    icon: <TagsOutlined />,
    type: 'lesson',
    downloadtitle: 'Télécharger',
    cloudtitle: 'Envoyer',
    deletetitle: 'Supprimer',
  };

  const card = (
    <Card {...textContent} />
  );

  it('Snapshot card', () => {
    const { baseElement } = render(card);
    expect(baseElement).toHaveTextContent("Minerva McGonagall");
  });
});