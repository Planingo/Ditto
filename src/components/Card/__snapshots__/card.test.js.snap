// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component Snapshot card 1`] = `
<body>
  <div>
    <div
      class="ant-card ant-card-bordered ant-card-hoverable ant-card-default ant-card-type-lesson"
      cloudtitle="Envoyer"
      deletetitle="Supprimer"
      downloadtitle="Télécharger"
      icon="[object Object]"
      src="https://avatars.bugsyaya.dev/285/"
      tags="Métamorphose,Ancient Runes,Arithmancy"
    >
      <div
        class="ant-card-head"
      >
        <div
          class="ant-card-head-wrapper"
        >
          <div
            class="ant-card-head-title"
          >
            Minerva McGonagall
          </div>
        </div>
      </div>
      <div
        class="ant-card-body"
      />
    </div>
  </div>
</body>
`;
